packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -113,3 +113,7 @@ HostPlatform getCurrentHostPlatform() {
 
   return HostPlatform.linux_x64;
 }
+
+String getBuildDirectory() {
+  return config.getValue('build-dir') ?? 'build';
```

Comment:
I think of the `flutter.yaml` as the project configuration instead of the build configuration. So I am not sure we should put the build-dir there.